src/scripts/settings.js

```diff
@@ -0,0 +1,167 @@
+    }
+    else {
+        empty.classList.add('hidden');
+        table.classList.remove('hidden');
+    }
+}
+emptyStateFunction(currentSettings);
+
+
+
+
+// JSON FROM HTML FILE
+// const initialState = document.querySelector(selectors.stateJson).textContent;
+// const state = JSON.parse(initialState);
+
+
+document.addEventListener('click', (event) => {
+    if (event.target.classList.contains(selectors.deleteFormula)) {
+        const parent = event.target.parentElement.parentElement;
+        parent.remove();
+        emptyStateFunction(currentSettings);
+    }
+});
+
+
+// EDIT
+document.addEventListener('click', (event) => {
+    if (event.target.classList.contains(selectors.pen)) {
+        if (document.querySelector(selectors.textInputforCreatingFormula).value == "") {
+            console.log("Write something ...");
+            alert("Write something ...");
+        }
+        else{
+            const sibling = event.target.parentElement.parentElement.firstElementChild.firstElementChild;
+            sibling.innerText = document.querySelector(selectors.textInputforCreatingFormula).value;
+            document.querySelector(selectors.textInputforCreatingFormula).value = "";
+        }
+    }
+});
+
+
+
+
+
+
+
+
+const field = document.querySelector(selectors.formulasFolder);
+const template = document.querySelector(selectors.template);
+// CREATE FORMULA
+document.addEventListener('click', (event) => {
+    if (event.target.classList.contains(selectors.createFormula) || event.target.classList.contains(selectors.createFormulaFromEmptyState)) {
+        if (document.querySelector(selectors.textInputforCreatingFormula).value == "") {
+            console.log("Write something ...");
+            alert("Write something ...");
+        }
+        else{
+            const newElement = document.createElement("tr");
+            newElement.classList.add("." + selectors.template);
+            newElement.innerHTML = template.innerHTML;
+            newElement.querySelector("." + selectors.link).innerText = document.querySelector(selectors.textInputforCreatingFormula).value;
+        
+            console.log(newElement);
+            field.appendChild(newElement);
+            document.querySelector(selectors.textInputforCreatingFormula).value = "";
+            emptyStateFunction(currentSettings);
+        }
+    }
+});
+
+
+
+const saveButton = document.querySelector(selectors.save);
+const discardButton = document.querySelector(selectors.discard);
+
+
+// SAVE
+saveButton.addEventListener('click', () => {
+    savedSettings.innerHTML = currentSettings.innerHTML;
+    console.log(savedSettings);
+})
+// DISCARD
+discardButton.addEventListener('click', () => {
+    document.querySelector(selectors.formulasFolder).innerHTML = savedSettings.innerHTML;
+    console.log(document.querySelector(selectors.formulasFolder));
+    emptyStateFunction(currentSettings);
+})
+
+
+
+// URL MANIPULATION
+document.addEventListener('click', (e) => {
```

Comment:
This can use useRoute() inside the component, I guess this will be more intuitive